iceoryx_utils/doc/lockfree_queue/lockfree_queue.md

```diff
@@ -0,0 +1,140 @@
+# Lock Free queue 

```

Comment:
Add copyright notice?

Reply:
It is not a source code file but can be done.